### ISSUE TITLE
Type error: Cannot read property data of null

### ISSUE THREAD

**Issue by mjanssen** [cristianbote/goober#142]
Getting an error thrown by both Preact & React with the following message:

`Uncaught TypeError: Cannot read property 'data' of null`

I have tried to debug this issue, but it seems like there's something going on deeper in Goober. I believe the issue is caused here, in this minified file:

<img width="649" alt="Screenshot 2020-05-07 at 14 41 30" src="https://user-images.githubusercontent.com/1145427/81295784-40239780-9071-11ea-9952-3c7b3fa6d299.png">

When I reload the page, everything seems to be fine again, so it doesn't happen every render 🤔 
Any help is appreciated :)

Cheers!

**Comment by mjanssen** [comment]
Additional info: 

The component I'm rendering is using the `css` method coming from Goober and it's loaded from an external library.

**Comment by lucat1** [comment]
If you are aware of the specific `css` call that is throwing this error, providing it would help.
That said, the error should come from [here](https://github.com/cristianbote/goober/blob/master/src/core/update.js#L20). That function then gets called by `hash` which itself is called from `css`. So yeah, the issue should come from a `css` call.

What I believe could be the issue is `getSheet` returning `null`. In fact, if you try to log `window._goober` that should be null, whereas in a working environment it refers to goober's stylesheet.
![Screenshot 2020-05-07 at 15 29 06](https://user-images.githubusercontent.com/29304787/81300090-824fd780-9077-11ea-8562-caf17a968ed5.png)

A few things that may help fix this issue:
- remove the `<style>` element from the page's HTML if you aren't doing SSR or the element itself is empty, as that tag is created by goober itself when not found.
- This verbose usage may also fix the issue:
```js
import { css } from 'goober'

css.call({ target: document.head }, `
  color: red;
`)

```

The best way to figure this out tough is to provide a reproducible example in CodeSandbox or some other replr, or at least the faulty code.

**Comment by mjanssen** [comment]
Hmm, thought I found a fix, but nope :(

I created a repo with the same setup. I'm using an external component, but also included it in the project itself (Grid.js).

Steps to reproduce

1. `npm i`
2. `npm run dev`
3. Go to `localhost:3000` (You should now see an empty page with something that looks like a table)
4. Click the about link
5. Refresh
6. Click on the home link

Switching back and forth from the index page to the about page should be all good.

Repo: https://github.com/mjanssen/nextjs-goober

**Comment by mjanssen** [comment]
Just running

```
css.call(
  { target: document.head },
  `
  color: red;
`
);
```
In a useEffect() defined in `about.js` gives the same error. Might be a problem with how Goober is initialised in Next at this moment 🤔.

**Comment by lucat1** [comment]
Thank you so very much for your reproduction guide. I've tried it and indeed it errors. I've nailed it down to the fact that the About page does not contain any calls to goober. In fact if I add any goober code at all on that page it doesn't error anymore. I'll dig deeper and see why this happens!

**Comment by mjanssen** [comment]
As far as I see, the problem lies in the fact that the <style> tag is empty upon first render.
When I change
`const css = extractCss();`
to 
` const css = `${extractCss()}.something{color:red;}`;`

Everything seems to be functional again.

**Comment by lucat1** [comment]
Ahh yes you got it right. The style element exists but its `firstChild` is null. [That's why](https://github.com/cristianbote/goober/blob/master/src/core/get-sheet.js#L22) it errors.

**Comment by cristianbote** [comment]
Heyo! @mjanssen thanks for opening this issue.

TL;DR;
goober expects a textNode to exist inside the style tag. Meaning in here https://github.com/mjanssen/nextjs-goober/blob/master/pages/_document.js#L21 you should do `html: ' ' + this.props.css`.

The `textNode` exists so goober can avoid the reflow/repaint from updating the contents, via a `textNode.data` property update. This one is a really odd one TBH, and tried before to move to `insertRule` which could lead to better perf, but in goober's case the metrics were the same.

I think this should be documented somewhere, or at least part of the `next-with-goober` example somewhere.

**Comment by mjanssen** [comment]
Ahhh that makes a lot of sense! Thanks for the explanation @cristianbote!
Thanks for checking with me @lucat1! I think we can close this 'issue' now ;)
